android-components/components/service/contile/src/main/java/mozilla/components/service/contile/ContileTopSitesProvider.kt

```diff
@@ -44,14 +45,17 @@ internal const val CACHE_TOP_SITES_KEY = "tiles"
  */
 class ContileTopSitesProvider(
     context: Context,
+    val isMozillaOnline: Boolean = false,
```

Comment:
Since this is in Android Components and intended to be a generic component, I would suggest renaming this to not signal anything related to our internal Mozilla use cases. Something like `hasExclusions` with the kDoc description indicating the JSON response has a list of hosts to exclude to parse.

There's also another way we could approach this if we know this list of hosts are fairly static and could just live within Fenix. We could just add a lambda to filter the list of hosts in https://searchfox.org/mozilla-mobile/source/firefox-android/android-components/components/feature/top-sites/src/main/java/mozilla/components/feature/top/sites/TopSitesConfig.kt#36. This would simplify everything if this is an okay approach for your use case.

However, if the list of hosts are fairly dynamic and is preferred to be fetch through this JSON response, let me know.

Reply:
The default top sites we put are the ones that we cooperate relatively stable and long lasting, while the partners of the special collaboration during shopping festival might be some of them, which may cause duplicate. But the special collaboration partners might change from some to others from time to time, thus I think we need a dynamic exclusions to hide different default top sites every shopping fes. It won't be frequent though, about twice a year. @gabrielluong

Reply:
Okay, could you make the suggested changes and we could do another review on the changes